app/page.tsx

```diff
@@ -57,6 +60,38 @@ export default function HomePage() {
       }),
   });
 
+  const handleSearch = (query: string) => {
+    const currentQuery = new URLSearchParams(window.location.search);
+    query ? currentQuery.set("s", query) : currentQuery.delete("s");
+    setSearch(query);
+    router.push(`${window.location.pathname}?${currentQuery.toString()}`);
+  };
+
+  const selectFilter = (option: FilterOption) => {
+    setSelectedOption(option);
+    setPopoverOpen(false);
+
+    const currentQuery = new URLSearchParams(window.location.search);
+    currentQuery.set("f", option.orderBy);
+    router.push(`${window.location.pathname}?${currentQuery.toString()}`);
+  };
+
+  useEffect(() => {
+    const searchParams = new URLSearchParams(window.location.search);
+    const searchValueFromSearchParams = searchParams.get("s");
+    const filterValueFromSearchParams = searchParams.get("f");
+
+    if (searchValueFromSearchParams) setSearch(searchValueFromSearchParams);
+
+    if (filterValueFromSearchParams) {
+      const matchedOption = filterOptions.find(
+        option => option.orderBy === filterValueFromSearchParams
+      );
+
+      if (matchedOption) setSelectedOption(matchedOption);
+    }
+  }, []);
```

Comment:
If we can remove this logic and have search params initialising `search` and `selectedOption` that would be better.